Assembler.UnitTests/FrameHandlers/EndFrameHandlerTests.cs

```diff
@@ -0,0 +1,144 @@
+            _dateTimeProviderMock.VerifyNoOtherCalls();
+        }
+
+        [Test]
+        public void Handle_IdentifierThrowsException_FrameNotBeingUsed()
+        {
+            // Arrange
+            var frame = new Mock<BaseFrame>(AssemblingPosition.Final);
+
+            _identifierFactoryMock.Setup(identifierFactory => identifierFactory.Create(It.IsAny<BaseFrame>()))
+                .Throws<NullReferenceException>();
+
+            // Act
+            Assert.DoesNotThrow(() => _handler.Handle(frame.Object));
+
+            // Assert
+            _identifierFactoryMock.Verify(identifier => identifier.Create(It.IsAny<BaseFrame>()), Times.Once);
+            _identifierFactoryMock.Verify(identifier => identifier.Create(frame.Object), Times.Once);
+        }
+
+        [Test]
+        public void Handle_IdentifierInCache_MessageBeingRemovedEnrichedAndReleased()
+        {
+            // Arrange
+            var frame = new Mock<BaseFrame>(AssemblingPosition.Final);
+            var message = new Mock<BaseMessageInAssembly>();
+
+            _cacheMock.Setup(cache => cache.Exists(It.IsAny<string>())).Returns(true);
+            _cacheMock.Setup(cache => cache.Get<BaseMessageInAssembly>(It.IsAny<string>())).Returns(message.Object);
+
+            // Act
+            _handler.Handle(frame.Object);
+
+            // Assert
+            _identifierFactoryMock.Verify(identifier => identifier.Create(It.IsAny<BaseFrame>()), Times.Once);
+            _identifierFactoryMock.Verify(identifier => identifier.Create(frame.Object), Times.Once);
+
+            _cacheMock.Verify(cache => cache.Exists(It.IsAny<string>()), Times.Once);
+            _cacheMock.Verify(cache => cache.Exists(_identifierString), Times.Once);
+
+            _cacheMock.Verify(cache => cache.Get<It.IsAnyType>(It.IsAny<string>()), Times.Once);
+            _cacheMock.Verify(cache => cache.Get<It.IsAnyType>(_identifierString), Times.Once);
+
+            _cacheMock.Verify(cache => cache.Remove(It.IsAny<string>()), Times.Once);
+            _cacheMock.Verify(cache => cache.Remove(_identifierString), Times.Once);
+
+            _enricherMock.Verify(enricher => enricher.Enrich(It.IsAny<BaseFrame>(), It.IsAny<BaseMessageInAssembly>()),
+                Times.Once);
+            _enricherMock.Verify(enricher => enricher.Enrich(frame.Object, message.Object), Times.Once);
+
+            _dateTimeProviderMock.Verify(provider => provider.Now, Times.Once);
+            Assert.AreEqual(DateTime.MinValue, message.Object.LastFrameReceived);
+
+            _messageReleaserMock.Verify(
+                releaser => releaser.Release(It.IsAny<BaseMessageInAssembly>(), It.IsAny<ReleaseReason>()), Times.Once);
+            _messageReleaserMock.Verify(
+                releaser => releaser.Release(message.Object, ReleaseReason.EndReceived), Times.Once);
+        }
+
+        [Test]
+        public void Handle_IdentifierNotInCacheAndShouldDispatchSingleFrame_MessageBeingCreatedEnrichedAndReleased()
```

Comment:
`AndShouldDispatchSingleFrame` - Not relevant anymore (remove from the name of the test)

Reply:
f